src/Datagrid/Pager.php

```diff
@@ -47,8 +47,8 @@ public function countResults(): int
     public function init(): void
     {
         $query = $this->getQuery();
-        if (null === $query) {
-            throw new \LogicException('The pager need a query to be initialised');
+        if (!$query instanceof ProxyQueryInterface) {
```

Comment:
This was changed recently.
But we still need the execute method to return a countable value.